port/unix/omrsock.c

```diff
@@ -550,7 +550,19 @@ omrsock_accept(struct OMRPortLibrary *portLibrary, omrsock_socket_t serverSock,
 int32_t
 omrsock_send(struct OMRPortLibrary *portLibrary, omrsock_socket_t sock, uint8_t *buf, int32_t nbyte, int32_t flags)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	int32_t bytesSent = 0;
+
+	if (NULL == sock || 0 >= nbyte) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+
+	bytesSent = send(sock->data, buf, nbyte, flags);
+	if(-1 == bytesSent) {
+		portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_SOCK_SEND_FAILED);
+		return OMRPORT_ERROR_SOCK_SEND_FAILED;
```

Comment:
@rwy0717 Do you mean setting the set_last_error as OMRPORT_ERROR_SOCK_SEND_FAILED with errno and then return -1 (because it is supposed to return the number of bytes sent)?

Reply:
Oh I see, as in we don't set the error and just return -1, indicating that it returned -1 number of bytes?

Reply:
Yep, that's what I'm thinking

Reply:
Oh sorry, I missed your last reply! I mean yes to the first thing, you should set the last error and return -1.

Reply:
Sounds good, I will do that :+1: Thank you Robert!